addon/components/x-toggle/template.hbs

```diff
@@ -1,19 +1,26 @@
-{{#if showLabels}}
-	<span class="toggle-text toggle-prefix {{size}}">
-		{{offLabelValue}}
-	</span>
-{{/if}}
+<span class="x-toggle-component">
+  {{#if showLabels}}
+  	<span class="toggle-text toggle-prefix off-label" onclick={{action 'setToValue' false}}>
+  		{{_offLabel}}
+  	</span>
+  {{/if}}
+  <span
+    id={{elementId}}
+    class="x-toggle-container {{size}}{{if toggled ' x-toggle-container-checked'}}{{if disabled ' x-toggle-container-disabled'}}"
+    onclick={{action 'onClick'}}
+  >
+    {{input id=forId type='checkbox' checked=toggled class='x-toggle' disabled=disabled name=name}}
```

Comment:
Well looking at this again one thing we should definitely do is move to a native HTML input control rather than using Ember's helper as native is several orders of magnitude faster.